src/re2_internal.ml

```diff
@@ -340,6 +367,21 @@ let%test_module _ = (module struct
 
 end)
 
+let%test_module _ = (module struct
+  let%test_unit _ = begin
+    match Dict.make [("^X",   `Starts_X);
+                     ("^Y",   `Starts_Y);
+                     (".*X",  `Contains_X);
+                     (".*Y",  `Contains_Y);
+                     (".*X$", `Ends_X);
+                     (".*Y$", `Ends_Y)] with
+    | Some d -> [%test_eq: [ `Starts_X | `Starts_Y | `Contains_X | `Contains_Y | `Ends_X | `Ends_Y] list]
+                    [] (List.filter (fun x -> not (List.mem x [`Starts_X; `Contains_X; `Contains_Y; `Ends_Y]))
+		    (Dict.get_all d "XY")
+    | _      -> failwith "failed to compile dictionary"
+  end
+end)
+
```

Comment:
Makes sense. I'll add tests for these cases. The existing test suite for regexes doesn't really cover these cases though.